aws/resource_aws_ssm_parameter.go

```diff
@@ -124,24 +179,45 @@ func resourceAwsSsmParameterPut(d *schema.ResourceData, meta interface{}) error
 	log.Printf("[INFO] Creating SSM Parameter: %s", d.Get("name").(string))
 
 	paramInput := &ssm.PutParameterInput{
-		Name:      aws.String(d.Get("name").(string)),
-		Type:      aws.String(d.Get("type").(string)),
-		Value:     aws.String(d.Get("value").(string)),
-		Overwrite: aws.Bool(d.Get("overwrite").(bool)),
+		Name:           aws.String(d.Get("name").(string)),
+		Type:           aws.String(d.Get("type").(string)),
+		Value:          aws.String(d.Get("value").(string)),
+		Overwrite:      aws.Bool(shouldUpdateSsmParameter(d)),
+		AllowedPattern: aws.String(d.Get("allowed_pattern").(string)),
+	}
+
+	if description, ok := d.GetOk("description"); ok {
+		paramInput.SetDescription(description.(string))
```

Comment:
We should probably avoid using the `Set...()` methods of the AWS SDK as they are likely going away in v2 of the SDK: `paramInput.Description = description.(string)`

That said, this is always writing a value to the description parameter and should be able to match similar code of other resources:

```go
if d.HasChange("description") {
  _, n := d.GetChange("description")
  paramInput.Description = aws.String(n.(string))
}
```

In the case where its removed from the configuration, it should still write "". Just something to consider.